ntoskrnl/ps/apphelp.c

```diff
@@ -450,9 +450,9 @@ ApphelpCacheInitialize(VOID)
     {
         ExInitializeResourceLite(&ApphelpCacheLock);
         RtlInitializeGenericTableAvl(&ApphelpShimCache,
-                                     ApphelpShimCacheCompareRoutine,
-                                     ApphelpShimCacheAllocateRoutine,
-                                     ApphelpShimCacheFreeRoutine,
+                                     (PRTL_AVL_COMPARE_ROUTINE)ApphelpShimCacheCompareRoutine,
+                                     (PRTL_AVL_ALLOCATE_ROUTINE)ApphelpShimCacheAllocateRoutine,
+                                     (PRTL_AVL_FREE_ROUTINE)ApphelpShimCacheFreeRoutine,
```

Comment:
These functions do not have correct prototypes already (and therefore you are forced to explicitly cast types here°??

Reply:
Yeah the signature looks right to me. Not sure where this comes from.

Reply:
I checked that before casting. I'm sure @HBelusca will tell us why this happens :)

Reply:
It is because the prototype defines the first argument as `struct _RTL_AVL_TABLE *Table`, and I defined it as `PRTL_AVL_TABLE Table`
notice the absence of `struct`

Reply:
And this was done possibly because the "typedef ... *PRTL_AVL_TABLE;" comes after (and not before) the declaration of these function types.